File: Seminars/src/main/java/org/example/SeminarOOP5/hw/model/Dog.java
package org.example.SeminarOOP5.hw.model;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
/**
 * Класс Dog расширяет класс Animal дополнительными полями, доступными для псов
 */
public class Dog extends Animal {
    private int weight;
    private String color;

    public Dog(String name, int weight, String color) {
        super(name);
        this.weight = weight;
        this.color = color;
    }

    @Override
    public String toString() { return String.format("Кличка пса: %s, Вес: %d, Окрас: %s", super.getName(), weight, color); }
}

File: Seminars/src/main/java/org/example/SeminarOOP7/hw/Dog.py
from Animal import Animal


class Dog(Animal):
    def __init__(self, nickname, weight,):
        super().__init__(nickname)
        self._weight = weight

    @property
    def weight(self):
        return self._weight

    @weight.setter
    def weight(self, new_weight):
        self._weight = new_weight

    def __str__(self) -> str:
        return f"{self.nickname} {self._weight}"